app/src/main/java/com/github/se/icebreakrr/ui/map/MapView.kt

```diff
@@ -80,18 +99,59 @@ fun HeatMap(
   var heatmapProvider by remember { mutableStateOf<HeatmapTileProvider?>(null) }
   var isMapLoaded by remember { mutableStateOf(false) }
   var lastCameraPosition by remember { mutableStateOf<LatLng?>(null) }
+  var isHeatmapVisible by remember { mutableStateOf(true) }
+
+  // Observe the meetingRequestChosenLocalisation
+  val meetingRequestChosenLocalisation =
+      myProfile.value?.meetingRequestChosenLocalisation ?: emptyMap()
+
+  // Create a list of UIDs to fetch profiles
+  val uidsToFetch = meetingRequestChosenLocalisation.keys.toList()
+
+  // Create a mutable list to hold the fetched profiles
+  val profilesMeeting = remember { mutableStateListOf<Profile>() }
+
+  // Fetch profiles for the UIDs
+  LaunchedEffect(uidsToFetch) {
```

Comment:
Problem: mutableStateListOf is used for profilesMeeting, which can lead to memory bloat if not cleaned up properly.
Solution: Clear profilesMeeting when it’s no longer needed

Reply:
You are right, I'll add some profiles cleaning in the code!